jaeger-spark-dependencies-cassandra/src/test/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJobTest.java

```diff
@@ -25,59 +26,85 @@
-        .withCreateContainerCmdModifier(cmd -> {
-          cmd.withLinks(new Link(cassandra.getContainerId(), "cassandra"));
-          cmd.withHostName("test_driver");
-        })
-        .withExposedPorts(14268, 16686, 8080, 9411);
-    jaegerTestDriver.start();
-    queryUrl = String.format("http://localhost:%d", jaegerTestDriver.getMappedPort(16686));
-    collectorUrl = String.format("http://localhost:%d", jaegerTestDriver.getMappedPort(14268));
-    zipkinCollectorUrl = String.format("http://localhost:%d", jaegerTestDriver.getMappedPort(9411));
+    jaegerCassandraSchema = new GenericContainer<>("jaegertracing/jaeger-cassandra-schema:latest")
+        .withNetwork(network);
+    jaegerCassandraSchema.start();
+    /**
+     * Wait until schema is created
+     */
+    await().until(() -> !jaegerCassandraSchema.isRunning());
+
+    jaegerCollector = new GenericContainer<>("jaegertracing/jaeger-collector:latest")
+        .withNetwork(network)
+        .withEnv("CASSANDRA_SERVERS", "cassandra")
+        .withEnv("CASSANDRA_KEYSPACE", "jaeger_v1_dc1")
+        .withEnv("COLLECTOR_ZIPKIN_HTTP_PORT", "9411")
+        .withEnv("COLLECTOR_QUEUE_SIZE", "100000")
+        .waitingFor(Wait.forHttp("/").forStatusCode(204))
+        // the first one is health check
+        .withExposedPorts(14269, 14268, 9411);
+    jaegerCollector.start();
+
+    jaegerQuery = new GenericContainer<>("jaegertracing/jaeger-query:latest")
+        .withNetwork(network)
+        .withEnv("CASSANDRA_SERVERS", "cassandra")
+        .withEnv("CASSANDRA_KEYSPACE", "jaeger_v1_dc1")
+        .waitingFor(Wait.forHttp("/").forStatusCode(204))
+        .withExposedPorts(16687, 16686);
+    jaegerQuery.start();
+
+    queryUrl = String.format("http://localhost:%d", jaegerQuery.getMappedPort(16686));
+    collectorUrl = String.format("http://localhost:%d", jaegerCollector.getMappedPort(14268));
+    zipkinCollectorUrl = String.format("http://localhost:%d", jaegerCollector.getMappedPort(9411));
   }
 
   @AfterClass
   public static void afterClass() {
     Optional.of(cassandra).ifPresent(GenericContainer::close);
-    Optional.of(jaegerTestDriver).ifPresent(GenericContainer::close);
+    Optional.of(jaegerCollector).ifPresent(GenericContainer::close);
+    Optional.of(jaegerQuery).ifPresent(GenericContainer::close);
+    Optional.of(jaegerCassandraSchema).ifPresent(GenericContainer::close);
   }
 
   @After
   public void after() {
     try (Cluster cluster = cassandra.getCluster(); Session session = cluster.newSession()) {
-      session.execute("TRUNCATE jaeger.traces");
-      session.execute("TRUNCATE jaeger.dependencies");
-
+      session.execute("TRUNCATE jaeger_v1_dc1.traces");
+      session.execute("TRUNCATE jaeger_v1_dc1.dependencies");
     }
   }
 
   @Override
   protected void deriveDependencies() throws Exception {
-    // flush all data to disk
-    cassandra.execInContainer("nodetool", "flush", "jaeger");
```

Comment:
What is the reason this is no longer required?

Reply:
it's redundant. It should query also mem tables. https://docs.datastax.com/en/cassandra/3.0/cassandra/dml/dmlAboutReads.html